docs/concepts.md

```diff
@@ -18,10 +18,27 @@ Your Meltano Cloud Organization is the top-level entity associated with your pay
 >
 > Meltano Cloud users grant access to specific public and private repositories using the standard GitHub App authorization flow.
 
+### Internal Organization ID
+
+Meltano Cloud uses an internal alpha-numeric string (randomly generated) to uniquely identify your cloud organization's identify. This ID may occassionally be shared during Cloud debugging and you may find references to it in the internal workings of Meltano Cloud CLI.
+
+> **Warning**
+> The "Internal Organization ID" should not be considered a permanent identifier, and may change at any time without notice.
+
 ## Meltano Cloud Projects
 
 A "Cloud Project" is any Meltano project you have registered on Meltano Cloud. The project definition consists of a git repo and a relative project directory within the git repo.
 
+### Internal Project ID
+
+Meltano Cloud uses an internal alpha-numeric string (randomly generated) to uniquely identify your project's deployment. This ID may occassionally be shared during Cloud debugging and you may find references to it in the internal workings of Meltano Cloud CLI.
+
+> **Warning**
+> The "Internal Project ID" should not be confused with the Project ID that is defined within `meltano.yml`.
+
+> **Note**
+> The "Internal Project ID" should not be considered a permanent identifier, and may change at any time without notice.
```

Comment:
If not by using the Cloud project ID, how does one specify which Cloud project they're operating on?

Reply:
For now, if we have to provide the internal project ID at runtime, I think that's totally acceptable. But if the login/logout verb is relative to a project, this makes life easier for the end user, while still being explicit.

Reply:
Going to ping @tayloramurphy here too because this is a significant design decision.

My understanding was that you logged into your Cloud user, who belongs to some organizations which have Meltano Cloud projects under them. After logging in, you can operate on any of the projects which you have access to.

Reply:
For the context of this discussion, I'll revert to `cloud_project_id` as the identifier here. Partly, I think this discussion is popping up now due to my need to disambiguate project ID from _cloud_ project ID. I'm not tied to calling it "internal" but we do need to be careful to keep the context explicit in our conversations, and in our docs for users.

We've discussed a few flows here:

1. `meltano cloud login` and (in the future) something like `meltano cloud project list|set` are separate commands.
   - Miro board has a possible syntax but this is not finalized: https://miro.com/app/board/uXjVP7UAEaM=/
   - ![image](https://user-images.githubusercontent.com/18150651/225126756-75232118-4b6d-48b8-bf01-93634cc75138.png)
1. `meltano cloud login` performs the equivalent to `meltano cloud project set` during the login process and therefor your login is _always_ contextual to a project. The cloud project ID can be:
   1. automatic if there is only one project deployed, 
   2. interactively selected if tty is available and no specific project is requested, or 
   3. explicitly requested via a `--project-id` or `--cloud-project-id` param.

Regardless, whether the internal project ID is stored during `login` or `project set`, once logged in, you would not need to explicitly provide the key - but the CLI _will_ have the key and will be able to provide it on all API calls.

Reply:
I've left some thoughts in the discussion AJ opened on this. I think we'll need a more human-readable project-name, but we can set limits on what it can contain (i.e. only letters, numbers, hyphens, and underscores).